honcho-cli/src/honcho_cli/commands/session.py

```
@@ -0,0 +1,407 @@
+    """Create or get a session."""
+    handle_cmd_flags(json_output=json_output, workspace=workspace)
+    sid = validate_resource_id(session_id, "session")
+    client, config = get_client()
+
+    parsed_metadata = None
+    if metadata:
+        try:
+            parsed_metadata = json.loads(metadata)
+        except json.JSONDecodeError as e:
+            print_error("INVALID_JSON", f"--metadata must be valid JSON: {e}", {})
+            raise typer.Exit(1)
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)